test/with_tmp_db

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+usage() {
+    cat << EOF
+
+USAGE: $0 <command>
+
+Runs the given command with the POSTGREST_TEST_CONNECTION environment variable
+set to a temporary database that is ready for running the PostgREST test suite.
+
+You'll need to have the Postgres binaries 'initdb' and 'pg_ctl' on your PATH.
+
+Example:
+
+    $0 stack test
+
+EOF
+    exit -1
+}
+
+if [ "$#" -lt 1 ]; then
+    echo "Please provide a command to be run with the temporary database."
+    usage
+fi
+
+# All data will be stored in a temporary directory.
+tmpdir="$(mktemp -d)"
+
+trap "rm -rf tmpdir" exit
+
+export PGDATA="$tmpdir"
+export PGHOST="$tmpdir"
+export PGUSER=postgrest_test_authenticator
+export PGDATABASE=postgres
+
+# Initialize a database cluster. We try to make it as independent as possible
+# from the host by specifying the timezone, locale and encoding.
+TZ=UTC initdb --no-locale --encoding=UTF8 --nosync -U "$PGUSER"
+
+# Start the database cluster. Instead of listening on a local port, we will
+# listen on a unix domain socket. We will reuse the $PGDATA directory for that
+# socket.
+pg_ctl start -o "-F -c listen_addresses=\"\" -k $PGDATA"
+
+stop() {
+    pg_ctl stop -m i
+    rm -rf "$tmpdir"
+}
+
+trap stop exit
+
+# Prepare the database for running the tests.
+psql <<EOF
+    create extension pgcrypto;
+    alter database $PGDATABASE set request.jwt.claim.id = '-1';
+EOF
```

Comment:
There is some other stuff that `test/create_test_db` did:

https://github.com/PostgREST/postgrest/blob/be674eb41d958716e9b4fa0fe186208992f73cb8/test/create_test_db#L50-L58

Do you think adding that would be unnecesary? IIRC, the `ALTER ROLE postgrest_test_authenticator SET default_text_search_config TO english; ` was necessary for users that run in another locale. Otherwise tests failed.

Reply:
I'm not sure about the default_text_search_config setting, I'll add it just in case!

The `LC_` settings should all be taken care of by the `--no-locale` flag given to `initdb`, which [sets everything to `C`](https://www.postgresql.org/docs/12/app-initdb.html), which in turn is [equivalent to `POSIX`](https://www.postgresql.org/docs/12/locale.html).